src/bindings/swig/ruby/kdb.i

```diff
@@ -0,0 +1,691 @@
+ * cases this is useful, to allow passing in Symbols as meta names. */
+%typemap(in) (const std::string & metaName) {
+  // typemap in for getMeta
+  $input = rb_funcall($input, rb_intern("to_s"), 0, NULL);
+  $1 = new std::string(StringValueCStr($input));
+}
+%typemap(freearg) (const std::string & metaName) {
+  // typemap in for getMeta
+  delete $1;
+}
+
+/* Typemap for setBinary
+ * pass raw data pointer of a Ruby String and its length */
+%typemap(in) (const void * newBinary, size_t dataSize) {
+  $1 = (void *) StringValuePtr($input);
+  $2 = RSTRING_LEN($input);
+}
+
+
+/* 'imitate' va_list as Ruby Hash
+ * 
+ * "missuse" the exception feature of SWIG to provide a custom
+ *  method invocation. This allows us to pass a Ruby argument hash
+ *  as a va_list. This way, we can imitate the variable argument
+ *  list (and keyword argument) features
+ */
+%typemap(in) (va_list ap) {
+  // we expect to be $input to be a Ruby Hash
+  Check_Type($input, T_HASH);
+}
+
+%feature("except") kdb::Key::Key (const char *keyName, va_list ap) {
+  /* standard method invocation would be: 
+  $action
+  */
+  /* exception features do not have local variables,
+     so we define them our selfs */
+  int hash_size = 0;
+  VALUE keys_arr;
+  VALUE key;
+  VALUE val;
+  int i;
+  int flags = 0;
+
+  /* $input substitution does not here, so we have to reverence
+     input variables directly */
+
+  hash_size = NUM2INT(rb_funcall(argv[1], rb_intern("size"), 0, NULL));
+  keys_arr = rb_funcall(argv[1], rb_intern("keys"), 0, NULL);
+  if (hash_size > 0) {
+    /* first we check if we can find the "flags" key.
+       this has to be passed to the kdb::Key constructor already */
+    for (i = 0; i < hash_size; i++) {
+      key = rb_ary_entry(keys_arr, i);
+      val = rb_hash_aref(argv[1], key);
+      /* convert key to String, in case of being a Symbol */
+      key = rb_funcall(key, rb_intern("to_s"), 0, NULL);
+      /* check for flags and extract them */
+      if (strcmp("flags", StringValueCStr(key)) == 0) {
+        Check_Type(val, T_FIXNUM);
+        flags = NUM2INT(val);
+        //printf("got flags: %d\n", flags);
+      }
+    }
+  }
+  /* invoke method
+     since we can't use arg2 here (is of type va_list)
+     we have to do it ourself (not very portable)
+  */
+  try {
+    result = (kdb::Key *)new kdb::Key((char const *)arg1, 
+      KEY_FLAGS, flags,
+      KEY_END);
+  } catch (std::bad_alloc &_e) {
+    SWIG_exception_fail(SWIG_MemoryError, (&_e)->what());
+  }
+  DATA_PTR(self) = result;
+  
+  if (hash_size > 0) {
+    /* now treat (nearly) all key-value pairs as meta data, thus
+       assign it to the newly created kdb::Key object */
+    for (i = 0; i < hash_size; i++) {
+      key = rb_ary_entry(keys_arr, i);
+      val = rb_hash_aref(argv[1], key);
+      key = rb_funcall(key, rb_intern("to_s"), 0, NULL);
+      val = rb_funcall(val, rb_intern("to_s"), 0, NULL);
+      /* ignore certain keys */
```

Comment:
actually, I've looked into this function and I think these two are the only reasonable here.
- `size` doesn't make sense, since Rubys `setBinary` doesn't take a size parameter (Ruby uses Strings for binary values and Strings have a `size` method).
- `func` is currently not implemented
- `meta`: everything except `value` or `flags` is considered as metadata
- other flags will be passed through `flags` key
- and the others (`KEY_NAME`, `KEY_OWNER`...) is considered as deprecated